include/onnxruntime/core/framework/kernel_def_builder.h

```diff
@@ -53,7 +53,13 @@ class KernelDef {
     return provider_type_;
   }
 
+  // type constraints with types supported by default
   const std::map<std::string, std::vector<MLDataType>>& TypeConstraints() const {
```

Comment:
this function is used here:
https://github.com/microsoft/onnxruntime/blob/b07e168a2b358e10423a29501bf49634281e6139/onnxruntime/python/onnxruntime_pybind_state.cc#L1009
checked with @jywu-msft and that usage is expecting the types supported by default